src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -48,8 +49,16 @@ Copyright (c) .NET Foundation. All rights reserved.
           DependsOnTargets="PrepareForBuild;CoreGenerateAssemblyInfo"
           Condition="'$(GenerateAssemblyInfo)' == 'true'" />
 
+  <Target Name="AddSourceRevisionToInformationalVersion"
+          DependsOnTargets="InitializeSourceControlInformation"
+          Condition="'$(IncludeSourceRevisionInInformationalVersion)' == 'true'">
+    <PropertyGroup>
+      <InformationalVersion>$(InformationalVersion) (revision $(SourceRevisionId))</InformationalVersion>
```

Comment:
Ah, yes, forgot to add SourceRevisionId != '' to the condition. 
It's set by InitializeSourceControlInformation target that is provided by a source control package.

Reply:
We should consider doing nothing if informational version already contains revision id. This allows custom formatting and will play better with pre existing techniques like nb.gv, right?

Reply:
How do we detect that though? Look for `+` in `InformationalVersion`?

Reply:
I was thinking $(InformationalVersion.Contains($(SourceRevisionId))

But that isn't foolproof due to ordering and possible SHA abbreviation.

Reply:
> But that isn't foolproof due to ordering and possible SHA abbreviation.

Yup, that's the problem.

Reply:
I thought about nb.gv interop as well (thanks, @nguerrera). It won't be a problem because nb.gv already turns generation of this attribute off completely to avoid conflicts with the one it generates itself.